python/ray/worker.py

```diff
@@ -900,7 +901,22 @@ def _wait_for_and_process_task(self, task):
                 "task_id": task.task_id().hex()
             }
             with profiling.profile("task", extra_data=extra_data, worker=self):
-                self._process_task(task, execution_info)
+                try:
```

Comment:
I think the `try`/`finally` is not necessary because `_process_task` function will handle all exceptions internally, so it won't raise any exceptions.